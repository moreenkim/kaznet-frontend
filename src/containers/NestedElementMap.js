import React, { Component } from 'react';
import ElementMap from './ElementMap';

export default class NestedElementMap extends Component {
  render() {
    const { detailitems } = this.props;
    const details = [];
    let counter = 1;

    for (const [key, value] of Object.entries(detailitems)) {
      details.push(
        React.createElement('tr', { key }, [
          <ElementMap
            key={counter}
            items={[key]}
            HTMLTag={this.props.HTMLTag}
            className="kaznet-title-td"
          />,
          // Need the key to be different as such we are multiplying it by 20 from the first one.
          <ElementMap key={counter * 20} items={[value]} HTMLTag={this.props.HTMLTag} />
        ])
      );
      counter++;
    }

    return details;
  }
}
