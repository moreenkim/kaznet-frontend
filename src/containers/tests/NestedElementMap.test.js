// Test NestedElement Map
import React from "react";
import { shallow, mount } from "enzyme";
import NestedElementMap from "../NestedElementMap";
import toJson from "enzyme-to-json";

describe("containers/NestedElementMap", () => {
  it("renders without crashing", () => {
    const detailItems = {
      Task: "Test",
      Amount: "4000 KES"
    };

    shallow(<NestedElementMap detailitems={detailItems} HTMLTag="td" />);
  });

  it("renders elements correctly", () => {
    const detailItems = {
      Task: "Test",
      Amount: "300034"
    };

    const wrapper = mount(
      <NestedElementMap detailitems={detailItems} HTMLTag="td" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
