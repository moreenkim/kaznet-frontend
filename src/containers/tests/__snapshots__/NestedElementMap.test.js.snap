// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/NestedElementMap renders elements correctly 1`] = `
<NestedElementMap
  HTMLTag="td"
  detailitems={
    Object {
      "Amount": "300034",
      "Task": "Test",
    }
  }
>
  <tr
    key="Task"
  >
    <ElementMap
      HTMLTag="td"
      className="kaznet-title-td"
      items={
        Array [
          "Task",
        ]
      }
      key="1"
    >
      <td
        className="kaznet-title-td"
        key="0"
      >
        Task
      </td>
    </ElementMap>
    <ElementMap
      HTMLTag="td"
      items={
        Array [
          "Test",
        ]
      }
      key="20"
    >
      <td
        key="0"
      >
        Test
      </td>
    </ElementMap>
  </tr>
  <tr
    key="Amount"
  >
    <ElementMap
      HTMLTag="td"
      className="kaznet-title-td"
      items={
        Array [
          "Amount",
        ]
      }
      key="2"
    >
      <td
        className="kaznet-title-td"
        key="0"
      >
        Amount
      </td>
    </ElementMap>
    <ElementMap
      HTMLTag="td"
      items={
        Array [
          "300034",
        ]
      }
      key="40"
    >
      <td
        key="0"
      >
        300034
      </td>
    </ElementMap>
  </tr>
</NestedElementMap>
`;
